saleor/graphql/discount/tests/queries/test_promotions_filtering.py

```diff
@@ -254,3 +256,74 @@ def test_query_promotions_filter_by_is_old_sale(
     assert {promotion_list[index].name for index in indexes} == {
         promotion["node"]["name"] for promotion in promotions
     }
+
+
+@pytest.mark.parametrize(
+    "filter_type",
+    [
+        PromotionTypeEnum.CATALOGUE,
+        PromotionTypeEnum.ORDER,
+    ],
+)
+def test_query_promotions_filter_type_eq(
+    filter_type, promotion_list, staff_api_client, permission_manage_discounts
+):
+    # given
+    promotion_with_order_type = promotion_list[:2]
+    Promotion.objects.filter(
+        id__in=[promotion.pk for promotion in promotion_with_order_type]
+    ).update(type=PromotionTypeEnum.ORDER.value)
+
+    input_type = filter_type.name
+    db_type = filter_type.value
+
+    variables = {"where": {"type": {"eq": input_type}}}
+
+    # when
+    response = staff_api_client.post_graphql(
+        QUERY_PROMOTIONS, variables, permissions=(permission_manage_discounts,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    promotions = content["data"]["promotions"]["edges"]
+    db_promotions = Promotion.objects.filter(type=db_type)
+    assert len(promotions) == db_promotions.count()
+    names = {node["node"]["name"] for node in promotions}
+    assert names == {promotion.name for promotion in db_promotions}
+
+
+@pytest.mark.parametrize(
+    "filter_type",
+    [
+        [PromotionTypeEnum.CATALOGUE],
+        [PromotionTypeEnum.ORDER],
+        [PromotionTypeEnum.ORDER, PromotionTypeEnum.ORDER],
```

Comment:
It's on purpose or do you want to provide two different types in the last element?

Reply:
@IKarbowiak Yes, I just wanted to confirm that it works when provided more that one type. I know that this doesn't make sense currently, but maybe in the future we will extend the types to have more than two types 😅 
If you think it doesn't make sense, I can revert it.